Makefile

```diff
@@ -45,11 +49,59 @@ $(wasm_out): $(wasm_src) $(sdk_srcs)
 	@echo "Building WASM module..."
 	@tinygo build -o $@ -target=wasi $<
 
-test_scripts := $(shell realpath $(wildcard ./test/scripts/*.txtar))
+# We manage the rendered template directory explicitly as the rendered templates
+# may contain secrets. Use `/tmp` if `TMPDIR` is not set.
+rendered_template_dir := $(or $(TMPDIR), /tmp)/bky-basm-rendered
+
+.PHONY: make-rendered-template-dir
+make-rendered-template-dir:
+	$(info rendered template directory: $(rendered_template_dir))
+	@mkdir -p $(rendered_template_dir)
+
+.PHONY: delete-rendered-template-dir
+delete-rendered-template-dir:
+	@rm -rf $(rendered_template_dir)
+
+# Specify all the files that are templates to be rendered
+template_src := $(wildcard $(template_dir)/*.mustache)
+# Specify all the files that we expect to exist after rendering as renames of the template files
+template_out := $(patsubst $(template_dir)/%.mustache,$(rendered_template_dir)/%,$(template_src))
+# Mark the rendered files as intermediate. Make will delete them after the build
+# if they still exist.
+.INTERMEDIATE: $(template_out)
+
+# Set default values for BASM_ variables. These can be overridden by the user
+# e.g. `BASM_USER_SECRET=mysecret make test`
+BASM_PLATFORM ?= plain
+BASM_CODE_MEASURE ?= plain
+BASM_AUTH_TOKEN ?= auth token
+BASM_HOST ?= local-server
+
+# Render each template from its mustache source. This rule is used when make sees
+# a dependency that matches the `$(rendered_template_dir)/%` pattern.
+# Syntax notes: `$<` represents the first dependency of the rule, which is the
+# mustache template file name. `$@` represents the target of the rule, which is
+# the rendered file name.
+$(rendered_template_dir)/%: $(template_dir)/%.mustache | make-rendered-template-dir
+	@jq -n \
+		--arg BASM_PLATFORM "$(BASM_PLATFORM)" \
+		--arg BASM_CODE_MEASURE "$(BASM_CODE_MEASURE)" \
+		--arg BASM_AUTH_TOKEN "$(BASM_AUTH_TOKEN)" \
+		--arg BASM_HOST "$(BASM_HOST)" \
+		'$$ARGS.named' \
+		| mustache $< > $@
+
+# Run the integration test and delete the temp dir afterward
+.PHONY: test-integration
+test-integration: $(wasm_out) $(template_out)
+	@txtar-c $(rendered_template_dir) \
+		| cat "$(script_dir)/integration.txtar" - \
+		| testscript -e WASM_FILE=$(wasm_out)
+	@$(MAKE) -s delete-rendered-template-dir
```

Comment:
This will not run if the previous command fails. However, I think that's ok since the rendered template directory is named  and will get deleted on a successful run. I felt like adding other mechanisms that ensured cleanup might have added a little more to this makefile than we really needed. Open to suggestions though!

Reply:
I think that is okay.  Trying to solve these types of problem with just make tends to result in more problems than are solved.  If we are really concerned with cleanup, we may want to look at more complex tools.  If we do want to "punt" on it, I wonder if it would be better to default to something like `$HOME/tmp/` instead of `/tmp`.  Even though it could stick around on a developer or test machine, it would at least live somewhere that is not accessible to everyone with access to the developer's computer.  Thoughts?

Reply:
I am mostly concerned with secrets sticking around on disk rather than temp files. The temp files, in the case that make crashes or something like that, should eventually get cleaned up by the OS if we are using TMPDIR, so I wasn't worried about losing some obsolete files on rare occasion.

Reply:
all a moot point now. I used txtar to allow for all input files to be specified together in one archive and rendered in one go with mustache. All in memory, no rendered files to disk.